interop/test_utils.go

```diff
@@ -454,6 +454,49 @@ func DoCancelAfterFirstResponse(tc testpb.TestServiceClient) {
 	}
 }
 
+// DoStatusCodeAndMessage checks that the status code is propagated back to the client.
+func DoStatusCodeAndMessage(tc testpb.TestServiceClient) {
+	var code int32 = 2
+	msg := "test status message"
+	respStatus := &testpb.EchoStatus{
+		Code:    &code,
+		Message: &msg,
+	}
+	// Test UnaryCall.
+	req := &testpb.SimpleRequest{
+		ResponseStatus: respStatus,
+	}
+	_, err := tc.UnaryCall(context.Background(), req)
+	if grpc.Code(err) != codes.Code(code) {
+		grpclog.Fatalf("/TestService/UnaryCall RPC compleled with error code %d, want %d", grpc.Code(err), code)
+	}
+	if err.Error() != fmt.Sprint("rpc error: code = ", code, " desc = ", msg) {
+		grpclog.Fatal("/TestService/UnaryCall unexpected RPC error message: ", err)
+	}
+	// Test FullDuplexCall.
+	stream, err := tc.FullDuplexCall(context.Background())
+	if err != nil {
+		grpclog.Fatalf("%v.FullDuplexCall(_) = _, %v", tc, err)
```

Comment:
github hid part of my comments. Please take a look at my original comment again. In idiomatic golang test, we typically add "want <nil>" if this function call does not expect any error.